docs/source/example_notebooks/gcm_counterfactual_medical_dry_eyes.ipynb

```diff
@@ -0,0 +1,291 @@
+   "source": [
+    "## Answering Alice's counterfactual querries\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "In cases where we want to examine a hypothetical outcome if an event had not happened or if it had happened\n",
+    "differently, we employ the so called Counterfactual logic based on structural causal models. Thus, we examine what the value of the target node (Vision) would be, if the Treatment node had a different value (0: for no treatment, or 1: for option A drops), given that we know the outcome of one Treatment value (in this case we know the target Vision for Treatment 2: the option B eye drops that Alice got), and given that the node Condition is 1 based on the Alices history (Alice has reported that she has the rare allergy). To check these counterfactual querries someone needs the aforementioned causal graph and to provide her own data, as well as the value of the Treatment that she wants to examine."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "counterfactual_data1 = gcm.counterfactual_samples(causal_model,\n",
+    "                                                  {'Treatment': lambda x: 1},\n",
+    "                                                  observed_data = specific_patient_data)\n",
+    "                                                   \n",
+    "counterfactual_data2 = gcm.counterfactual_samples(causal_model,\n",
+    "                                                  {'Treatment': lambda x: 0},\n",
+    "                                                  observed_data = specific_patient_data)\n",
+    "    \n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "df_plot2 = pd.DataFrame()\n",
+    "df_plot2['Vision after option B'] = specific_patient_data['Vision']\n",
+    "df_plot2['Counterfactual vision (option A)'] = counterfactual_data1['Vision']\n",
+    "df_plot2['Counterfactual vision (No treatment)'] = counterfactual_data2['Vision']\n",
+    "\n",
+    "df_plot2.plot.bar(title=\"Counterfactual outputs\")\n",
+    "plt.xlabel('Alice')\n",
+    "plt.ylabel('Eye sight quality')\n",
+    "plt.legend()    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "What Alice saw was that if she had taken option 'A' instead, her Vision would have become worse (Vision after taking A would be less if she had taken B). Therefore she realised that the rare condition (Condition=1) that she had reported in the App could be the one that would have caused an alergic reaction to the popular option 'A'. Alice was also able to see that if she had not taken any of the recommended options, she would have resulted with worse eye sight (the vaariable Vision resulted in a smaller relative value) than with option 'B' that she took."
```

Comment:
I'm just getting up to speed on the GCM models.  This violation of monotonicity is interesting.  Is there a reference for the fitted SCM?  Thanks.